src/main/java/org/wikivoyage/listings/validators/LongitudeValidator.java

```diff
@@ -1,16 +1,40 @@
 package org.wikivoyage.listings.validators;
 
+import org.apache.commons.lang.StringUtils;
 import org.wikivoyage.listings.entity.Listing;
 
 public class LongitudeValidator extends SimpleValidator {
-    @Override
-    public void validate(Listing poi) {
-        if (poi.getLongitude() != null && !poi.getLongitude().equals("")) {
-            try {
-                Float.parseFloat(poi.getLongitude());
-            } catch (NumberFormatException e) {
-                poi.add(ValidationIssue.INVALID_LONGITUDE);
-            }
-        }
-    }
+    
+	//0 at the prime meridian; 180 degrees east and west
+  	private static final int MIN_LONGITUDE = -180;
+  	private static final int MAX_LONGITUDE = 180;
+  	
+      @Override
+      public void validate(Listing poi) {
+      	String longitude = poi.getLongitude();
+      	if (StringUtils.isEmpty(longitude)) {
+      		//Do nothing; not invalid but no point parsing
+      		return;
+      	}
+      	
+      	//Sanitise the longitude to see if it contained invalid chars. Not trying to fix, just flag as invalid.
+      	// This is more robust than just relying on the NFE because we explicitly know what a longitude SHOULD contain.
+      	String sanitisedLongitude = longitude.replaceAll("[^0-9.-]","");
+      	if (!sanitisedLongitude.equals(longitude)) {
+          	poi.add(ValidationIssue.INVALID_LONGITUDE);
+          }
+          else {
+          	//Otherwise try to parse this as a valid longitude.
+              try {
+                  float latitudeF = Float.parseFloat(longitude);
```

Comment:
You might want to change the variable name to match the actual meaning. 👍

Reply:
*groan* Good spot on the variable name. Caused by copy-pasting, so a good sign that refactoring is needed, should be easy enough.